Hystrix-clj IllegalStateException when using collapsers

Using a very simple example:

```
(defcommand foo [a] (+ 1 a))
(defcollapser fooz (collapse [arg-lists] (instantiate foo (map first arg-lists))) (map [arg-lists batch-result] batch-result))
```

Calling `(fooz [1])` raises:

```
java.lang.RuntimeException: HystrixCollapser$0 HystrixCollapser failed while executing.
    at com.netflix.hystrix.HystrixCollapser.execute(HystrixCollapser.java:302)
    at com.netflix.hystrix.core.proxy$com.netflix.hystrix.HystrixCollapser$0.execute(Unknown Source)
    at com.netflix.hystrix.core$execute.doInvoke(core.clj:465)
    at clojure.lang.RestFn.applyTo(RestFn.java:139)
    at clojure.core$apply.invoke(core.clj:619)
...
Caused by: java.lang.IllegalStateException: HystrixRequestContext.initializeContext() must be called at the beginning of each request before RequestVariable functionality can be used.
    at com.netflix.hystrix.strategy.concurrency.HystrixRequestVariableDefault.get(HystrixRequestVariableDefault.java:79)
    at com.netflix.hystrix.strategy.concurrency.HystrixRequestVariableHolder.get(HystrixRequestVariableHolder.java:68)
    at com.netflix.hystrix.HystrixCollapser.getCollapserForUserRequest(HystrixCollapser.java:392)
    at com.netflix.hystrix.HystrixCollapser.queue(HystrixCollapser.java:321)
    at com.netflix.hystrix.core.proxy$com.netflix.hystrix.HystrixCollapser$0.queue(Unknown Source)
    at com.netflix.hystrix.HystrixCollapser.execute(HystrixCollapser.java:290)
    ... 30 more
```

--- a/hystrix-contrib/hystrix-clj/src/main/clojure/com/netflix/hystrix/core.clj
+++ b/hystrix-contrib/hystrix-clj/src/main/clojure/com/netflix/hystrix/core.clj
@@ -227,7 +227,7 @@
 
 ;################################################################################
 
-(defmacro with-context
+(defmacro with-request-context
   "Executes body within a new Hystrix Context"
   [& body]
   `(let [context# (HystrixRequestContext/initializeContext)]

--- a/hystrix-contrib/hystrix-clj/src/test/clojure/com/netflix/hystrix/core_test.clj
+++ b/hystrix-contrib/hystrix-clj/src/test/clojure/com/netflix/hystrix/core_test.clj
@@ -104,7 +104,7 @@
         (is (= (- 99 42) (.execute c)))))
 
     (testing "makes a HystrixCommand that implements getCacheKey"
-      (with-context
+      (with-request-context
         (let [call-count (atom 0) ; make sure run-fn is only called once
                test-def (normalize (assoc base-def
                                      :run-fn       (fn [arg] (swap! call-count inc) (str arg "!"))

